diff -Nur root_v5.34.18/core/base/src/TTimeStamp.cxx root_v5.34.18_patched/core/base/src/TTimeStamp.cxx
--- root_v5.34.18/core/base/src/TTimeStamp.cxx	2014-03-14 23:21:52.000000000 +0800
+++ root_v5.34.18_patched/core/base/src/TTimeStamp.cxx	2018-01-10 08:27:36.919000060 +0800
@@ -813,7 +813,7 @@
           tmstruct.tm_isdst);
 #if (defined(linux) && !defined(R__WINGCC)) || defined(R__MACOSX)
    printf(",\n      tm_gmtoff %6ld, tm_zone \"%s\"",
-#if defined(__USE_BSD) || defined(R__MACOSX)
+#if defined(__USE_BSD) || defined(R__MACOSX) || defined(__USE_MISC)
    tmstruct.tm_gmtoff, tmstruct.tm_zone);
 #else
    tmstruct.__tm_gmtoff, tmstruct.__tm_zone);
